polkadot-protocol/protocol-components/accounts/account-data-structure.md

```diff
@@ -0,0 +1,132 @@
+- `AccountData`: A configurable structure that can store various types of data specific to the account.
+
+## Account Reference Counters
+
+Account reference counters track dependencies in the runtime. For instance, if data is stored in a map associated with an account, you wouldn't want to delete the account until the data under its control has been removed.
+
+The `consumers` and `providers` reference counters are intended to work together. For example, in the [`session`](https://docs.rs/pallet-session/latest/pallet_session/) pallet, the consumers counter is incremented when an account sets its session keys before becoming a validator. 
+
+The providers counter must be greater than zero for the consumers counter to be incremented.
+
+### Providers Reference Counters
+
+The `providers` reference counter indicates whether an account is ready to be depended upon. For example, it is incremented when a new account is created with a balance greater than the existential deposit.
+
+The `providers` reference counter prevents Substrate pallets from storing data about an account until the account is active (`providers` > 0)
+
+### Consumers Reference Counters
+
+The `consumers` reference counter ensures that Substrate pallets don't remove an account until all associated data across pallets is deleted (that is, when `consumers` == 0). 
+
+It holds users accountable for the data they store on-chain. If users wish to delete their accounts and reclaim their existential deposit, they must first remove all data stored across all on-chain pallets to decrement the `consumers` counter.
+
+### Sufficients Reference Counter
+
+The `sufficients` reference counter indicates whether an account is self-sufficient and can exist independently. 
+
+For example, in the [`assets`](https://paritytech.github.io/polkadot-sdk/master/pallet_assets/index.html) pallet, an account may hold a sufficient quantity of specific assets without needing to maintain a native account balance.
+
+### Account Deactivation 
+
+Pallets also include cleanup functions that decrement the `providers` reference counter, marking the account as deactivated within the pallet-managed scope. 
+
+Once both the `providers` and `consumers` reference counters reach zero, the account is considered deactivated by all on-chain pallets.
+
+### Updating Counters
+
+Runtime developers can manage these counters using the `inc_consumers()`, `dec_consumers()`, `inc_providers()`, `dec_providers()`, `inc_sufficients()`, and `dec_sufficients()` methods provided by the `frame-system` pallet. 
+
+For every increment of a counter during an account's lifecycle, a corresponding decrement should be called to ensure proper management of the account state.
+
+There are also three query functions to ease usage on these counters:
+
+- `can_inc_consumer()` to check if an account is ready to be used (`providers` > 0).
```

Comment:
ditto, or link the functions in the docs

Reply:
Same, I added links to all these three functions from **frame_system::pallet::Pallet**